src/diagnostics/core_diagnostics.jl

```diff
@@ -38,24 +190,245 @@ function compute_relative_humidity!(
     # FIXME: Avoid extra allocations when ClimaCore overloads .= for this use case
     # We will want: out .= integrator.u.c.ρ
     thermo_params = CAP.thermodynamics_params(cache.params)
-    return TD.relative_humidity.(thermo_params, cache.ᶜts)
+    return TD.total_specific_humidity.(thermo_params, cache.ᶜts)
 end
 
-compute_relative_humidity!(out, state, cache, time) =
-    compute_relative_humidity!(
+compute_specific_humidity!(out, state, cache, time) =
+    compute_variable!(
+        Val(:specific_humidity),
         out,
         state,
         cache,
         time,
         cache.atmos.moisture_model,
     )
 
-# FIXME: Gabriele wrote this as an example. Gabriele doesn't know anything about the
-# physics. Please fix this!
 add_diagnostic_variable!(
-    short_name = "relative_humidity",
-    long_name = "relative_humidity",
-    units = "",
-    comments = "Relative Humidity",
-    compute! = compute_relative_humidity!,
+    short_name = "Specific_humidity",
+    long_name = "Specific Humidity",
+    units = "kg kg^-1",
+    comments = "Mass of all water phases per mass of air, a prognostic variable",
+    compute! = compute_specific_humidity!,
+)
+
+###
+# Surface specific humidity (2d) - although this could be collapsed into the 3d version + reduction, it probably makes sense to keep it separate, since it's really the property of the surface
+###
+function compute_variable!(
+    ::Val{:surface_specific_humidity},
+    out,
+    state,
+    cache,
+    time,
+    moisture_model::T,
+) where {T <: Union{EquilMoistModel, NonEquilMoistModel}}
+    # FIXME: Avoid extra allocations when ClimaCore overloads .= for this use case
+    # We will want: out .= integrator.u.c.ρ
+    thermo_params = CAP.thermodynamics_params(cache.params)
+    return TD.total_specific_humidity.(thermo_params, cache.sfc_conditions.ts)
+end
+
+compute_surface_specific_humidity!(out, state, cache, time) =
+    compute_variable!(
+        Val(:surface_specific_humidity),
+        out,
+        state,
+        cache,
+        time,
+        cache.atmos.moisture_model,
+    )
+
+add_diagnostic_variable!(
+    short_name = "surface_specific_humidity",
+    long_name = "Surface Specific Humidity",
+    units = "kg kg^-1",
+    comments = "Mass of all water phases per mass of air in the near-surface layer",
+    compute! = compute_surface_specific_humidity!,
+)
+
+###
+# Surface temperature (2d)
+###
+add_diagnostic_variable!(
+    short_name = "surface_temperature",
+    long_name = "Surface Temperature",
+    units = "K",
+    comments = "Temperature of the surface",
+    compute! = TD.air_temperature.(thermo_params, cache.sfc_conditions.ts),
```

Comment:
`compute!` has to take four specific arguments, so we have to wrap it around another function